idea/vara-eth/frontend/src/app/api/use-ethereum-client.ts

```diff
@@ -1,14 +1,30 @@
+import { useQuery } from '@tanstack/react-query';
 import { EthereumClient } from '@vara-eth/api';
 import { type WebSocketTransport } from 'viem';
-import { useWalletClient, usePublicClient } from 'wagmi';
+import { usePublicClient, useWalletClient } from 'wagmi';
 
-import { ROUTER_CONTRACT_ADDRESS } from '@/shared/config';
+import { ETH_CHAIN_ID, ROUTER_CONTRACT_ADDRESS } from '@/shared/config';
 
-export function useEthereumClient({ chainId }: { chainId?: number } = {}) {
-  const { data: walletClient } = useWalletClient({ chainId });
-  const publicClient = usePublicClient({ chainId });
+export function useEthereumClient() {
+  const { data: walletClient } = useWalletClient({ chainId: ETH_CHAIN_ID });
+  const publicClient = usePublicClient({ chainId: ETH_CHAIN_ID });
 
-  if (!walletClient || !publicClient) return null;
+  return useQuery({
+    queryKey: ['ethereumClient', ETH_CHAIN_ID, walletClient?.account?.address, publicClient?.chain?.id],
```

Comment:
```suggestion
    queryKey: ['ethereumClient', walletClient, publicClient],
```

`ETH_CHAIN_ID` is static, no need for it.

Is it possible to pass `walletClient` and `publicClient` as a whole? Both of them can contain some dynamic properties that can change and we will not be aware about it. The one that I encountered by myself - `walletClient` can have a different chain id until it's synced with `publicClient`.